src/wireviz/svgembed.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import re
+import base64
+from pathlib import Path
+
+mime_subtype_replacements = {'jpg': 'jpeg', 'tif': 'tiff'}
+
+def embed_svg_images(svg_in: str, base_path: Path):
+    # first, find any image references in SVG data, and cache the respective base64-encoded image
+    images_b64 = {}  # cache of base64-encoded images
+    re_xlink=re.compile(r"xlink:href=\"(?P<URL>.*?)\"", re.IGNORECASE)
+    for xlink in re_xlink.finditer(svg_in):
+        imgurl = xlink.group('URL')
+        if not imgurl in images_b64:  # only encode/cache every unique URL once
+            if not Path(imgurl).is_absolute():
+                imgurl_abs = (Path(base_path) / imgurl).resolve()
+            else:
+                imgurl_abs = Path(imgurl)
+
+            with open(imgurl_abs,'rb') as img:
+                img_bin = img.read()
+                img_b64 = base64.b64encode(img_bin)
+                img_str = img_b64.decode('utf-8')
+            images_b64[imgurl] = img_str
+    # second, replace links with the base64-encoded data
+    svg_out = svg_in
+    for url, b64 in images_b64.items():
+        svg_out = svg_out.replace(url, f'data:image/{get_mime_subtype(url)};base64, {b64}')
```

Comment:
A better alternative to the sorting above, is to include known characters before and after the URL:
```suggestion
        svg_out = svg_out.replace(f'href="{url}"', f'href="data:image/{get_mime_subtype(url)};base64, {b64}"')
```

Reply:
How about using only `f'="{url}"'`, without `href`, to avoid any upper/lower case issues?
The sorting by length using lambda is clever, but a bit cryptic; this approach seems more straightforward :)

Reply:
That is a valid point.
How about a Regex that picks out `href="*"` only within `<a *>`, to avoid the need for a parser as additional dependency?
Sure, a parser may work better for some ultra-rare edge case, but it's not worth it IMHO.

Reply:
How about a case-independent Regex that matches `xlink:href="*"`?
It would ignore a regular `<a href="*">` and wouldn't bloat the project with a parser as an additional dependency.
Sure, there might be ultra-rare cases that the Regex couldn't handle as well as a parser, but I'm willing to take that risk ;-)
What do you think?

Reply:
> How about a case-independent Regex that matches `xlink:href="*"`?
> It would ignore a regular `<a href="*">` and wouldn't bloat the project with a parser as an additional dependency.

They will look like `<a xlink:href="*">` and there might also be additional attributes in the same tag preceding or succeeding `xlink:href`. To find only `<image>` tags using regular expressions, I suggest searching for the full tag.

> Sure, there might be ultra-rare cases that the Regex couldn't handle as well as a parser, but I'm willing to take that risk ;-)
> What do you think?

See my rewrite suggestion to reduce the risk.

Reply:
Marking as resolved, following the regex approach below instead.